src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,177 @@
+            node.prev = null;
+            node.next = current.next;
+            head = node;
+        } else {
+            node.prev = current.prev;
+            node.next = current.next;
+            current.next.prev = node;
+            current.prev.next = node;
+        }
+        T oldItem = current.item;
+        return oldItem;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        check(index);
+        Node<T> current = findNodeByIndex(index);
+        if (current.equals(head)) {
+            head = head.next;
+        } else {
+            current.prev.next = current.next;
+        }
+        if (current.equals(tail)) {
+            tail = tail.prev;
+        } else {
+            current.next.prev = current.prev;
+        }
+        size--;
+        return current.item;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        boolean found = false;
+        Node<T> current = head;
+        for (int i = 0; i < size; i++) {
+            if (current.item == object || current.item != null && current.item.equals(object)) {
+                if (current == head) {
+                    head = current.next;
+                    if (head != null) {
+                        head.prev = null;
+                    }
+                } else if (current == tail) {
+                    tail = current.prev;
+                    tail.next = null;
+                } else {
+                    current.prev.next = current.next;
+                    current.next.prev = current.prev;
+                }
+                found = true;
+                size--;
+                break;
+            }
+            current = current.next;
+        }
+        return found;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
+        if (size == 0) {
+            return true;
+        }
         return false;
     }
+
+    public void check(int index) throws IndexOutOfBoundsException {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Invalid index");
+        }
+    }
+
+    private Node<T> findNodeByIndex(int index) {
+        check(index);
+        Node<T> node;
+        if (index < size / 2) {
+            node = head;
+            for (int i = 0; i < index; i++) {
+                node = node.next;
+            }
+        } else {
+            node = tail;
+            for (int i = size - 1; i > index; i--) {
+                node = node.prev;
+            }
+        }
+        return node;
+    }
+
+    private static class Node<T> {
+        private T item;
+        private Node prev;
+        private Node next;
+
+        public Node(Node prev, T item, Node next) {
+            this.prev = prev;
+            this.item = item;
+            this.next = next;
+        }
+
+        public Node(T item) {
+            this.item = item;
+        }
+    }
```

Comment:
Don't use getters and setters in class Node since we have access to private fields of the inner class. Also, private methods and classes should be after public ones in your class.